### PR TITLE
Mixin pattern fix

### LINKED ISSUE #26: Remove at-mixin-pattern rule for external use
# Issue Description

The rule in stylelint.config.js
```js
'scss/at-mixin-pattern': '^(terra-)[a-z]+([a-z0-9-]+[a-z0-9]+)?$',
```
Requires external users to preface their mixins with `terra-`

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior

No error should be thrown when not using the `terra-` prefix.

## Current Behavior

Running stylelint with a mixin without `terra-` causes an error to be thrown

## Steps to Reproduce

1. Create a mixin without the prefix of `terra-`
2. Run `npm run lint:scss`

## Environment

* Version: 2.0.0
* Browser Name and Version: N/A
* Operating System and version (desktop or mobile): MacOS Mojave

## @ Mentions

@bjankord

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Resolves #26

### CODE REVIEW

**Review by emilyrohrbough — APPROVED** [review]

**Comment by emilyrohrbough** [comment]
Please add yourself to the [Contributors](https://github.com/cerner/stylelint-config-terra/blob/master/CONTRIBUTORS.md) list 😄

File: stylelint.config.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = {
   ],
   rules: {
     'max-nesting-depth': 3,
-    'scss/at-mixin-pattern': '^(terra-)[a-z]+([a-z0-9-]+[a-z0-9]+)?$',
+    'scss/at-mixin-pattern': '^[a-z]+([a-z0-9-]+[a-z0-9]+)?$',
     'suitcss/custom-property-no-outside-root': true,
     'custom-property-pattern': [
       'terra-[a-z]+([a-z0-9-]+[a-z0-9]+)?$',
```
Comment (StephenEsser) [inline stylelint.config.js:19]: We may need to do something similar for this rule down the road.

**Review by StephenEsser — APPROVED** [review]

**Review by yuderekyu — APPROVED** [review]

**Review by bjankord — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,9 @@ ChangeLog
 
 Unreleased
 ----------
+### Changed
+* Removed Terra prefix from mixin pattern rule.
```
Comment (avinashg1994) [inline CHANGELOG.md]: This can be moved under the other "Changed" section.

**Review by avinashg1994 — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (saket2403) [reply CHANGELOG.md]: Resolved in 968992f

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ Unreleased
 
 ### Removed
 * Removed DangerJS integration
+* Removed Terra prefix from mixin pattern rule.
```
Comment (StephenEsser) [inline CHANGELOG.md]: This is probably being nit picky, but I think this belongs in the Changed section. We didn't remove any functionality, we changed the existing functionality.

Comment (saket2403) [reply CHANGELOG.md]: Resolved in 7346f40

**Review by bjankord — APPROVED** [review]

**Pull request merged**